pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -629,7 +629,7 @@ let
       clang =
         if stdenv.targetPlatform.libc == null then
           tools.clangNoLibc
-        else if stdenv.targetPlatform.useLLVM or false then
+        else if stdenv.targetPlatform.cc == "clang" && stdenv.targetPlatform.cxxlib == "libcxx" && stdenv.targetPlatform.bintools == "llvm" && stdenv.targetPlatform.rtlib == "compiler-rt" then
```

Comment:
Won’t this cause Darwin to use `clangUseLLVM`? Should there also be a check for `targetPlatform.linker == "lld"`?

Reply:
> Won’t this cause Darwin to use `clangUseLLVM`?

Yes, I'll add an exclusion

> Should there also be a check for `targetPlatform.linker == "lld"`?

I don't really think so once we get the linker attribute wired up correctly. It checks bintools at least which matters more for `clangUseLLVM`.